pkg/reconcilermanager/controllers/rootsync_controller.go

```diff
@@ -684,7 +684,8 @@ func (r *RootSyncReconciler) populateContainerEnvs(ctx context.Context, rs *v1be
 				r.reconcilerPollingPeriod.String(), rs.Spec.SafeOverride().StatusMode,
 				v1beta1.GetReconcileTimeout(rs.Spec.SafeOverride().ReconcileTimeout),
 				v1beta1.GetAPIServerTimeout(rs.Spec.SafeOverride().APIServerTimeout),
-				enableRendering(rs.GetAnnotations()),
+				enableAnnotation(metadata.RequiresRenderingAnnotationKey, rs.GetAnnotations()),
+				enableAnnotation(metadata.RequiresNSControllerAnnotationKey, rs.GetAnnotations()),
```

Comment:
Thinking about this a bit more, is it necessary to communicate between `reconciler` and `reconciler-manager` to set this variable? In the case of `RequiresRendering`, this communication is necessary to enable/disable the hydration-controller sidecar.

In this case we are toggling the namespace controller, which is an additional controller in the existing reconciler container. I'm wondering if we can enable/disable this controller without restarting the reconciler Pod, and by extension not require the annotation to communicate with `reconciler-manager`. WDYT?

Reply:
The namespace controller is registered in the controller-manager. When the controller-manager starts, it starts.
I we want to enable/disable it within the reconciler, we will need to restart the controller-manager, which seems to be feasible by listening to a channel. I'll give it a try.

Reply:
For the record I'm not altogether opposed to this abstraction. If this ends up being a cleaner abstraction than trying to restart the controller-manager while the reconciler is running, then I'm on board with this implementation.

Feel free to time box your investigation and please let me know what you find

Reply:
Thanks, will keep you posted.

Reply:
The controller-manager is shared by both the Finalizer controller and the namespace controller.
If a namespace controller is enabled/disabled, and the controller-manager is restarted, the finalizer will also be restarted, which might be okay because the reconciliation loop should be idempotent, although it still feels hacky to restart a controller-manager shared by multiple controllers.

That might be another reason why we would like to move the controller-manager to the top layer, so that its lifecycle go with the Pod container's lifecycle.

Reply:
@karlkfi  WDYT?